pkg/utils/utils_linux.go

```diff
@@ -91,9 +100,26 @@ func determineEndian() binary.ByteOrder {
 	return endian
 }
 
-// Get all the veth interfaces.
-// Similar to ip link show type veth
-func GetInterface(name string, interfaceType string) (netlink.Link, error) {
+// GetOutgoingInterface gets the outgoing interface by first trying
+// executing an equivalent to `ip route get <External IP>`
```

Comment:
I am curious about the use of the IP addresses above (1.1.1.1). While its common in k8s to just have a default route, it a multiple nic setup you could have explicit routes such as 1.1.1.0/24 in that case whatever dev is associated with that route would have the ebpf program attached to it. I believe this might not help as we would be most interested in the cluster traffic? Is it better to determine the interface based upon the result of the CNI or the default route such as 'ip route get default'

Reply:
I am also curious about multipathing as well. However that most likely would land in any work related to supporting multiple interfaces.